src/components/Search/Search.jsx

```diff
@@ -1,11 +1,21 @@
-import React from 'react';
+import React, { useState } from 'react';
+import PropTypes from 'prop-types';
 
 import './Search.modules.scss';
 
-export const Search = () => {
+export const Search = ({ details }) => {
+  const [searchMovie, setSearchMovie] = useState('');
+
+  const filteredMovie = details?.length
```

Comment:
You can revrite it like that:
```
  const filteredMovie = useMemo(() => {
    return details.filter((movie) => {
      return movie.name === searchMovie;
    });
  }, [details, searchMovie]);
```

1.  If details is an empty array there will be no one iteration of the `filter` function, so you do not need to check `details.length`
2. You coul use `useMemo` hook to rember calculated value and prevent redundunt rerenders if `details or searchMovie` was not changed